include/revng/Model/model-schema.yml

```diff
@@ -245,469 +349,915 @@ definitions:
+        type: uint64_t
+
+  - name: RelocationType
+    type: enum
+
+    doc: |-
+      A enumeration describing how a `Relocation` should be written at the target
+      address.
+
+    members:
+      - name: WriteAbsoluteAddress32
+        doc: Write the absolute address of the object as a 32-bit integer.
+
+      - name: WriteAbsoluteAddress64
+        doc: Write the absolute address of the object as a 64-bit integer.
+
+      - name: AddAbsoluteAddress32
+        doc: |-
+          Add to the 32-bit integer present at the target location the absolute
+          address of the object.
+
+      - name: AddAbsoluteAddress64
+        doc: |-
+          Add to the 64-bit integer present at the target location the absolute
+          address of the object.
+
+      - name: WriteRelativeAddress32
+        doc: |-
+          Write the address of the object as a 32-bit integer, expressed as a
+          relative from the target of the relocation.
+
+      - name: WriteRelativeAddress64
+        doc: |-
+          Write the address of the object as a 64-bit integer, expressed as a
+          relative from the target of the relocation.
+
+      - name: AddRelativeAddress32
+        doc: |-
+          Add to the 32-bit integer present at the target location the address of
+          the object, expressed as a relative from the target of the relocation.
+
+      - name: AddRelativeAddress64
+        doc: |-
+          Add to the 32-bit integer present at the target location the address of
+          the object, expressed as a relative from the target of the relocation.
+
+  - name: CallSitePrototype
+    type: struct
+
+    doc: Information about the prototype of a specific callsite within a `Function`.
+
+    key:
+      - CallerBlockAddress
+
+    fields:
+      - name: CallerBlockAddress
+        doc: Address of the basic block of the call.
+        type: MetaAddress
+
+      - name: Prototype
+        doc: The prototype specific to *this* call site.
+        type: Type
+        upcastable: true
+
+      - name: IsTailCall
+        doc: Whether this call site is a tail call or not.
+        type: bool
+        optional: true
+
+      - name: Attributes
+        doc: Attributes for this call site.
+        sequence:
+          type: MutableSet
+          elementType: FunctionAttribute
+        optional: true
+
+  - name: FunctionAttribute
+    type: enum
+
+    doc: |-
+      Attributes for function calls. They can either be applied to specific call
+      sites, or to *all* the call sites of a specific function.
+
+    members:
+      - name: NoReturn
+        doc: The function call does not return.
+
+      - name: Inline
+        doc: The function call must be inlined in the caller.
+
+  - name: StatementComment
+    type: struct
+
+    doc: |-
+      A comment that can be attached to a statement in the decompiled code
+      based on the debug information attached to the said statement uniquely
+      identifying it.
```

Comment:
The fact that we used debug information is an implementation detail, leave it out.

Reply:
What about
```
      A comment that can be attached to a statement in the decompiled code
      *if* we have a way to uniquely identify said statement.
```
?